### ISSUE TITLE
Hyperopt not using correct config 

### ISSUE THREAD

**Issue by xmatthias** [freqtrade/freqtrade#915]
## Describe your environment

  * Python Version: 3.6.5 (`python -V`)
  * Branch: Develop
  * Last Commit ID: d7e7ef11f9b7c38520432ec54daf6d5f95dcca3c
  
## Step 3: Describe the problem:

Running hyperopt with a custom config file (exchange binance) does not recognize correct folder.

### Steps to reproduce:

my configuration configures a few ETH pairs on binance.

when running hyperopt against it, it seems to completely ignore my configuration, run against BTC bittrex pairs (for which i don't have backtest-data).

#### config:

``` json
{
    "max_open_trades": 10,
    "stake_currency": "ETH",
    "stake_amount": 0.10,
    "fiat_display_currency": "USD",
    "dry_run": true,
    "unfilledtimeout": 600,
    "bid_strategy": {
        "ask_last_balance": 0.0
    },
    "exchange": {
        "name": "binance",
        "key": "your_exchange_key",
        "secret": "your_exchange_secret",
        "pair_whitelist": [
            "LTC/ETH",
            "IOTA/ETH",
	        "XMR/ETH",
            "XRP/ETH",
            "NCASH/ETH",
            "MTH/ETH",
            "AST/ETH",
            "TNT/ETH",
            "ETC/ETH"
        ],
        "pair_blacklist": [
        ]
    },
    "experimental": {
        "use_sell_signal": false,
        "sell_profit_only": false
    },
    "telegram": {
        "enabled": true,
	    "token": "546065661:AAGSqjM2F3Zajs_HpiZY5LAYfD3AWmaGM3Y",
	    "chat_id": "462034882"
    },
    "initial_state": "running",
    "internals": {
        "process_throttle_secs": 5
    }
}

```

#### logs: 

```bash
freqtrade --config config.binance_ETH_test_all.json hyperopt --timerange 20180601-20180605

2018-06-13 20:03:22,060 - freqtrade.configuration - INFO - Log level set to INFO
2018-06-13 20:03:22,061 - freqtrade.configuration - INFO - Dry run is disabled
2018-06-13 20:03:22,061 - freqtrade.configuration - INFO - Using DB: "sqlite:///tradesv3.sqlite"
2018-06-13 20:03:22,061 - freqtrade.configuration - INFO - Using max_open_trades: 3 ...
2018-06-13 20:03:22,061 - freqtrade.configuration - INFO - Parameter --timerange detected: 20180601-20180605 ...
2018-06-13 20:03:22,061 - freqtrade.configuration - INFO - Using data folder: user_data/data/bittrex ...
2018-06-13 20:03:22,061 - freqtrade.configuration - INFO - Parameter --epochs detected ...
2018-06-13 20:03:22,061 - freqtrade.configuration - INFO - Will run Hyperopt with for 100 epochs ...
2018-06-13 20:03:22,061 - freqtrade.configuration - INFO - Parameter -s/--spaces detected: all
2018-06-13 20:03:22,063 - freqtrade.strategy.resolver - INFO - Using resolved strategy DefaultStrategy from '/home/xmatt/development/python/cryptos/freqtrade/freqtrade/strategy'
2018-06-13 20:03:22,063 - freqtrade.strategy.resolver - INFO - Override strategy 'minimal_roi' with value in config file.
2018-06-13 20:03:22,063 - freqtrade.strategy.resolver - INFO - Override strategy 'stoploss' with value in config file: -0.1.
2018-06-13 20:03:22,063 - freqtrade.exchange - INFO - Instance is running with dry_run enabled
2018-06-13 20:03:22,064 - freqtrade.exchange - INFO - Using Exchange "Bittrex"
2018-06-13 20:03:23,640 - freqtrade.optimize - WARNING - No data for pair: "ETH/BTC", Interval: 5m. Use --refresh-pairs-cached to download the data
2018-06-13 20:03:23,640 - freqtrade.optimize - WARNING - No data for pair: "LTC/BTC", Interval: 5m. Use --refresh-pairs-cached to download the data
2018-06-13 20:03:23,640 - freqtrade.optimize - WARNING - No data for pair: "ETC/BTC", Interval: 5m. Use --refresh-pairs-cached to download the data
2018-06-13 20:03:23,640 - freqtrade.optimize - WARNING - No data for pair: "DASH/BTC", Interval: 5m. Use --refresh-pairs-cached to download the data
2018-06-13 20:03:23,640 - freqtrade.optimize - WARNING - No data for pair: "ZEC/BTC", Interval: 5m. Use --refresh-pairs-cached to download the data
2018-06-13 20:03:23,640 - freqtrade.optimize - WARNING - No data for pair: "XLM/BTC", Interval: 5m. Use --refresh-pairs-cached to download the data
2018-06-13 20:03:23,640 - freqtrade.optimize - WARNING - No data for pair: "NXT/BTC", Interval: 5m. Use --refresh-pairs-cached to download the data
2018-06-13 20:03:23,640 - freqtrade.optimize - WARNING - No data for pair: "POWR/BTC", Interval: 5m. Use --refresh-pairs-cached to download the data
2018-06-13 20:03:23,640 - freqtrade.optimize - WARNING - No data for pair: "ADA/BTC", Interval: 5m. Use --refresh-pairs-cached to download the data
2018-06-13 20:03:23,640 - freqtrade.optimize - WARNING - No data for pair: "XMR/BTC", Interval: 5m. Use --refresh-pairs-cached to download the data
```

**Comment by xmatthias** [comment]
The way hyperopt currently works is it loads the config from `user_data/hyperopt.py`.

using this i got hyperopt to work with my pairs.

**Comment by vertti** [comment]
This was fixed in #920.
